std/net/curl.d

```diff
@@ -627,6 +627,61 @@ unittest
     assert(res == cast(ubyte[])[17, 27, 35, 41]);
 }
 
+/* Encode associative array using www-form-urlencoding
+ *  perhaps this should live in std.uri
+ */
+private string enc()(in string[string] values)
+{
+    string enc;
+    foreach (k, v; values)
+    {
+        if (enc.length)
+            enc ~= '&';
+        import std.uri;
+        enc ~= encodeComponent(k) ~ "=" ~ encodeComponent(v);
+    }
+    return enc;
```

Comment:
You could use an _Appender!string_ here, particularly since you can reserve using _values.length_.